source/slang/slang-ir-specialize.cpp

```diff
@@ -1787,7 +1787,25 @@ struct SpecializationContext
                 // created.
                 //
                 auto valType = val->getFullType();
-                if (auto extractExistentialType = as<IRExtractExistentialType>(valType))
+                if (!isCompileTimeConstantType(valType))
```

Comment:
I wonder if we should make the fix more general, by just calling `createParam` here, but add another version of `cloneInstDecorationsAndChildren` that takes a list of parameters, and insert that list of parameters in the first block, before cloning any remaining insts.

Because we still have other branches in thie function that are making calls to `createParam` that theoretically can have the same hoisting problem.